test/mobx.js

```diff
@@ -332,4 +332,43 @@ describe('usage with mobx should generally work', () => {
     expect(tree.getNode(resultsNode.path).page).to.equal(1)
     expect(service).to.have.callCount(2)
   })
+  it('the markForUpdate traversal should not try to alter arrays', async () => {
+    // When a group node has the join "not" and is changed to something else,
+    // it triggers the reactor for "join", which then
+    // triggers the reactor of "all", which returns the children,
+    // which makes it so the array of children is sent to markForUpdate,
```

Comment:
let's update this comment with the new description of this PR since it's about something different now